### PR TITLE
Initial support for generating ASP.NET Core Controllers

### PR DESCRIPTION
**Pull request by kleferbe**
Generating ASP.NET Core controllers is a bit different in that the generated controller files become user code and will be carefully updated with new methods found in the API by caffoa, without changing or removing existing methods.

This way we can keep the advantages of all the binding magic of asp.net core and modify the method signatures and attributes as needed.

### CODE REVIEW

File: CdIts.Caffoa.Cli/Generator/ModelGenerator.cs [hunk 1]
```diff
@@ -322,7 +322,11 @@ private void WriteEnumPropertyClass(PropertyData property, string className)
         string jsonproperty = "";
         if (type == "string")
         {
-            jsonproperty = "StringEnumConverter";
+            jsonproperty = _config.Flavor switch
+            {
+                CaffoaConfig.GenerationFlavor.JsonNet => "StringEnumConverter",
+                CaffoaConfig.GenerationFlavor.SystemTextJson => "JsonStringEnumConverter",
+            };
```
Comment (claasd) [inline CdIts.Caffoa.Cli/Generator/ModelGenerator.cs]:
```suggestion
            {
                CaffoaConfig.GenerationFlavor.SystemTextJson => "JsonStringEnumConverter",
                _ => "StringEnumConverter",
             };
```

Comment (claasd) [reply CdIts.Caffoa.Cli/Generator/ModelGenerator.cs]: the default is null if nothing is given,

Comment (claasd) [reply CdIts.Caffoa.Cli/Generator/ModelGenerator.cs]: ... and default null means Json.NET

**Comment by claasd** [comment]
There is a lot of edge cases that are not yet working correctly.
Please add a Demo Project in demo/ folder (DemoAsp), and add generation for the controller to the caffoa.yml
```
  - apiPath: tests/base.openapi.yml
    config:
      prefix: Asp
      flavor: SystemTextJson
    model:
      namespace: DemoAsp.Model.Base
      targetFolder: ./demo/DemoAsp/Model/Base

  - apiPath: tests/openapi.yml
    config:
      prefix: Asp
      flavor: SystemTextJson
      imports:
        - DemoAsp.Model.Base
    model:
      namespace: DemoAsp.Model
      targetFolder: ./demo/DemoAsp/Model
    controller:
      targetFolder: demo/DemoAsp
      namespace: DemoAsp
      name: DemoAsp
```

You will find a lot of issues with includes and object naming.

* includes are not corrected if generated wrong initially
* object names do not use prefixes
* Json.NET inlcudes are generated

Also, you could easily use the inferface apprach, avoiding all the problems that come with manipulating existing user code (like includes, naming changes and so on).

**Review by claasd — CHANGES_REQUESTED** [review]

**Comment by claasd** [comment]
Looks good so far, could you write some key points into the readme? Or should be start with experimental mode?

**Review by claasd — APPROVED** [review]

**Comment by kleferbe** [comment]
Most things should now be working and whats not is mostly System,Text.Json model related.
Added a demo project for ASP.NET with Newtonsoft too.

**Review by claasd — APPROVED** [review]
